### PR TITLE
feat(day): isolate logic and render prop on the day level

### PR DESCRIPTION
**Pull request by PeterMK85**
### Description

+ Split default render from the logic.
+ Define isolated Render prop component for Day
+ Move date related functions into a standalone helper and expose them as props
+ Set these options as default props on parent component to reduce the checks of availability

### CODE REVIEW

File: src/RenderPropsComponents/WeekDay/test.tsx [hunk 1]
```diff
@@ -57,90 +59,80 @@ test('trigger callback on mouseenter', () => {
   expect(handleOnEnter).toBeCalled()
 })
 
-test('custom render', () => {
```
Comment (PeterMK85) [inline src/RenderPropsComponents/WeekDay/test.tsx]: No more if else, every sub component will be rendered by a render prop, and on the top most component, we will set these as default props

File: src/calendar/__specs__/month_spec.tsx [hunk 2]
```diff
@@ -16,7 +16,7 @@ describe('Month', () => {
     wrapper = shallow(<Month {...props} />)
     instance = wrapper.instance()
     mockEvent = {
-      currentTarget: { value: date },
+      currentTarget: { dataset: { simpleReactCalendarDay: date } },
```
Comment (PeterMK85) [inline src/calendar/__specs__/month_spec.tsx]: not sure, but currently `data-simple-react-calendar-day` where we are rely on some data.
Possbile `data-sc-date` or just `data-date`, WDYT?

File: src/calendar/week.tsx [hunk 3]
```diff
@@ -114,53 +123,52 @@ export default class Week extends Component<Props, {}> {
...
+      return renderWeekDay({
+        ISODate: getISODate(day),
+        blockClassName,
+        date: day,
+        getWeekDayFormatted,
+        handleOnClick: isSelectable
+          ? onDayClick
+          : isDisabled
+          ? onDisabledDayClick
+          : () => {},
```
Comment (aesqe) [inline src/calendar/week.tsx]: Maybe extract it into `const noop = () => {}`?

File: src/calendar/__specs__/month_spec.tsx (hunk 2 above)
Comment (havenchyk) [reply src/calendar/__specs__/month_spec.tsx]: Peter’s variant looks better to me

File: src/calendar/calendar.tsx [hunk 4]
```diff
@@ -278,13 +294,14 @@ export default class Calendar extends Component<Props, State> {
...
         customRender={renderMonth}
-        renderDay={renderDay}
+        renderWeekDay={renderWeekDay}
         renderWeek={renderWeek}
         renderDaysOfWeek={renderDaysOfWeek}
         renderDayOfWeek={renderDayOfWeek}
         activeMonth={this._activeMonth()}
         blockClassName={blockClassName}
         daysOfWeek={daysOfWeek}
+        getWeekDayFormatted={getWeekDayFormatted}
```
Comment (denieler) [inline src/calendar/calendar.tsx]: maybe those `getWeekDayFormatted` and `getISODate` can be parts of the `WeekDay` implementation? and if you write your own `renderWeekDay` then you can reuse them from `utils`? (not critical or something, just thoughts, so please ignore them if they are not relevant)

File: src/calendar/week.tsx (hunk 3 above)
Comment (havenchyk) [reply src/calendar/week.tsx]: fine to me

**Pull request merged**